### PR TITLE
Support for array:some langlib function

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Related to #35183

### CODE REVIEW

**Review by hasithaa — DISMISSED** [review]
Looks good to me.

**Comment by hasithaa** [comment]
@KavinduZoysa it seems like, Semantic API tests are failing. Please check.

File: langlib/lang.array/src/main/ballerina/array.bal [hunk 1]
```diff
@@ -119,6 +119,21 @@ public isolated function reduce(Type[] arr, @isolatedParam function(Type1 accum,
     name: "reduce"
 } external;
 
+# Tests whether a function returns true for some member of an array.
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply langlib/lang.array/src/main/ballerina/array.bal:122]: There can be minor differences between the spec docs and the implementation docs atm, to ensure docs are generated properly.

**Comment by KavinduZoysa** [comment]
> @KavinduZoysa it seems like, Semantic API tests are failing. Please check.

Changed

File: langlib/lang.array/src/main/ballerina/array.bal (hunk 1 above)
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply langlib/lang.array/src/main/ballerina/array.bal:122]: Done

File: langlib/lang.array/src/main/ballerina/array.bal [hunk 2]
```diff
@@ -119,6 +119,21 @@ public isolated function reduce(Type[] arr, @isolatedParam function(Type1 accum,
...
+# func` is called for each member of `arr` in order unless and until a call returns true.
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply langlib/lang.array/src/main/ballerina/array.bal]: Changed

File: langlib/lang.array/src/main/ballerina/array.bal [hunk 3]
```diff
@@ -119,6 +119,21 @@ public isolated function reduce(Type[] arr, @isolatedParam function(Type1 accum,
...
+# + arr - the array
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply langlib/lang.array/src/main/ballerina/array.bal:128]: Changed

File: langlib/lang.array/src/main/ballerina/array.bal [hunk 4]
```diff
@@ -119,6 +119,21 @@ public isolated function reduce(Type[] arr, @isolatedParam function(Type1 accum,
...
+# + func - function to apply to each member
+# + return - true if func returns true for some member of `arr`; otherwise, false
+#
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply langlib/lang.array/src/main/ballerina/array.bal]: Changed

**Comment by MaryamZi** [comment]
@KavinduZoysa, there seem to be test failures.

**Comment by gimantha** [comment]
Lets format the json files

**Review by gimantha — APPROVED** [review]
LGTM

File: langlib/lang.array/src/main/ballerina/array.bal [hunk 5]
```diff
@@ -119,6 +119,24 @@ public isolated function reduce(Type[] arr, @isolatedParam function(Type1 accum,
...
 
+# Tests whether a function returns true for some member of an array.
+#
+# The parameter `func` is called for each member of parameter `arr` in order unless and until a call returns true.
+# When the array is empty, returns false.
+#
+# + arr - the array
+# + func - function to apply to each member
+# + return - true if applying parameter `func` returns true for some member of `arr`; otherwise, false
+#
```
Comment (MaryamZi) [inline langlib/lang.array/src/main/ballerina/array.bal:130]:
```suggestion
```

File: langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java [hunk 6]
```diff
@@ -461,9 +461,37 @@ public void testArrayLibNegativeCases() {
...
+        CompileResult negativeResult = BCompileUtil.compile("test-src/arraylib_test_negative_semantic.bal");
+        int errorIndex = 0;
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "incompatible types: 'int' will not be matched to 'string'", 18, 28);
+        Assert.assertEquals(negativeResult.getErrorCount(), errorIndex);
+    }
+
+    @Test(expectedExceptions = BLangRuntimeException.class,
+            expectedExceptionsMessageRegExp = "error: \\{ballerina/lang.array}IndexOutOfRange " +
+                    "\\{\"message\":\"array index out of range: index: 2, size: 2\"}.*")
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java:489]: We can do these assertions in b7a itself. We can trap the error and validate the message and "message" field from the detail.

Comment (MaryamZi) [reply langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java:489]: We can get rid of `Assert.fail` also.

Comment (KavinduZoysa) [reply langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java]: Changed

File: langlib/lang.array/src/main/ballerina/array.bal (hunk 5 above)
Comment (KavinduZoysa) [reply langlib/lang.array/src/main/ballerina/array.bal]: Changed

File: langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal [hunk 7]
```diff
@@ -1458,3 +1458,103 @@ function testUnshiftLargeValues() {
...
+    assertValueEquality(arr.some(func4), true);
+}
+
+function func5(map<int> val) returns boolean {
+    int? ii = val["k"];
+    return ii !is () && ii > 10;
+}
+
+function testSome9() {
+    map<int>[] arr = [{i: 2, j: 3}, {i: 20, j: 30}];
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal:1558]: Are these called?

Comment (KavinduZoysa) [reply langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal:1558]: No, updated

**Review by gimantha — DISMISSED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/config/foreach_stmt_ctx_config12.json [hunk 8]
```diff
@@ -1,590 +1 @@
...
... ()` func: a function to apply to each member  \n  \n**Return** `array:Type1[]`   \n- new array containing result of applying parameter `func` to each member of parameter `arr` in order  \n  \n"}},"sortText":"CD","filterText":"'map","insertText":"'map(${1})","insertTextFormat":"Snippet","command":{"title":"editor.action.triggerParameterHints","command":"editor.action.triggerParameterHints"}},{"label":"some(function () func)","kind":"Function","detail":"boolean","documentation":{"right":{"kind":"markdown","value":"**Package:** _ballerina/lang.array:0.0.0_  \n  \nTests whether a function returns true for some member of an array.\n\nThe parameter `func` is called for each member of parameter `arr` in order unless and until a call returns true.\nWhen the array is empty, returns false.\n  \n**Params**  \n- `function ()` func: function to apply to each member  \n  \n**Return** `boolean`   \n- true if applying parameter `func` returns true for some member of `arr`; otherwise, false  \n  \n"}},"sortText":"CD","filterText":"some","insertText":"some(${1})","insertTextFormat":"Snippet","command":{"title":"editor.action.triggerParameterHints","command":"editor.action.triggerParameterHints"}}]}
```
Comment (MaryamZi) [inline language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/config/foreach_stmt_ctx_config12.json:1]: Needs to be formatted?

**Review by MaryamZi — APPROVED** [review]

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Comment by dulajdilshan** [comment]
Can you @nadeeshaan @IMS94 verify the test changes?

**Review by malinthar — APPROVED** [review]
LGTM

**Pull request merged**